Yoast/wordpress-seo
# Remove option to control showing date in google preview

## Context

* Removes the option (setting) to control if the date should be shown in the Google Preview. The date will now always be shown for post-type previews.
* Taxonomies never had the functionality and because they don't have a date themselves, this won't be shown on the taxonomy-Google Previews.

## Summary

This PR can be summarized in the following changelog entry:

* Removes the setting to show the date in the Google Preview, the date will now always be shown in post-type previews.

## Relevant technical choices:

* No upgrade routine needed: After saving the settings; the `showdate-*` options will disappear because they are no longer defined in `inc/options/class-wpseo-option-titles.php`
* Removed a method that is no longer used (cleanup, could have been done sooner)
* No changes are needed in the JavaScript repository, as there is no logic present there, besides checking for an empty value of the date to add the component.

## Test instructions
This PR can be tested by following these steps:

* Go to Yoast > Search Appearance > Content types
* See the setting "Date in Google Preview" no longer present
* Go to any Google Preview (Snippet Preview) and witness the date being displayed in the meta-description-area

## UI changes

## Documentation

## Quality assurance


Fixes P2-92
